core/src/main/java/org/kiwix/kiwixmobile/core/history/HistoryAdapter.kt

```diff
@@ -0,0 +1,149 @@
+import butterknife.ButterKnife
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.core.R.drawable
+import org.kiwix.kiwixmobile.core.R.layout
+import org.kiwix.kiwixmobile.core.R.string
+import org.kiwix.kiwixmobile.core.R2.id
+import org.kiwix.kiwixmobile.core.extensions.setBitmapFromString
+import org.kiwix.kiwixmobile.core.history.HistoryListItem.DateItem
+import org.kiwix.kiwixmobile.core.history.HistoryListItem.HistoryItem
+import org.threeten.bp.LocalDate
+import org.threeten.bp.format.DateTimeFormatter
+
+internal class HistoryAdapter(
+  val historyList: List<HistoryListItem>,
+  val deleteList: List<HistoryListItem>,
+  val itemClickListener: OnItemClickListener
+) : Adapter<ViewHolder>() {
+
+  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+    return if (viewType == TYPE_ITEM) {
+      Item(inflateLayoutFromParent(parent, layout.item_bookmark_history))
+    } else {
+      Category(inflateLayoutFromParent(parent, layout.header_date))
+    }
+  }
+
+  private fun inflateLayoutFromParent(parent: ViewGroup, layoutId: Int) =
+    LayoutInflater.from(parent.context)
+      .inflate(layoutId, parent, false)
+
+  override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+    if (holder is Item) {
+      val history = historyList[position] as HistoryItem
+      setItemDataWithHelpOfHistoryItem(holder, history)
+    } else if (holder is Category) {
+      val date = (historyList[position] as DateItem).dateString
+      setCategoryDataWithHelpOfDate(date, holder)
+    }
+  }
+
+  private fun setCategoryDataWithHelpOfDate(date: String, holder: Category) {
+    val formatter = DateTimeFormatter.ofPattern("d MMM yyyy")
+    val todaysDate = LocalDate.now().format(formatter)
+    val yesterdayDate = LocalDate.now().minusDays(1).format(formatter)
+
+    if (todaysDate != null && todaysDate.contentEquals(date)) {
+      holder.date.setText(
+        R.string.time_today
+      )
+    } else if (yesterdayDate != null && yesterdayDate.contentEquals(date)) {
+      holder.date.setText(
+        R.string.time_yesterday
+      )
+    } else {
+      holder.date.text = date
+    }
+  }
+
+  private fun setItemDataWithHelpOfHistoryItem(holder: Item, history: HistoryItem) {
+    holder.title.text = history.historyTitle
+    if (deleteList.contains(history)) {
+      holder.favicon.setImageDrawable(
+        ContextCompat.getDrawable(
+          holder.favicon.context,
+          drawable.ic_check_circle_blue_24dp
+        )
+      )
+    } else {
+      holder.favicon.setBitmapFromString(history.favicon)
+    }
+    holder.itemView.setOnClickListener {
+      itemClickListener.onItemClick(
+        holder.favicon, history
+      )
+    }
+    holder.itemView.setOnLongClickListener {
+      itemClickListener.onItemLongClick(
+        holder.favicon, history
+      )
+    }
+  }
+
+  override fun getItemViewType(position: Int): Int {
+    return if (historyList[position] is DateItem) 0 else TYPE_ITEM
```

Comment:
these one liners that return a value are good candidates for expression functions